### PR TITLE
Update builder to kind v0.4.0

### PR DESCRIPTION
**Pull request by howardjohn**
I will move to a real istio repo if this passes tests ok, adding hold for now

### CODE REVIEW

File: test/docker/Dockerfile [hunk 1]
```diff
@@ -13,16 +13,16 @@ FROM docker:latest as docker
 ############################################
 # Image to create go binaries
 FROM golang:1.12.5 as golang
-ENV     GO111MODULE=on
-ENV     GOPROXY=https://proxy.golang.org
```
Comment (costinm) [inline test/docker/Dockerfile]: Why did you remove the proxy ? I think it was supposed to speed things up and make sure deleted repos don't affect us.

File: test/docker/Dockerfile [hunk 2]
```diff
@@ -35,12 +35,12 @@ RUN chmod +x /usr/local/bin/repo
...
 # create istioctl from `master`, assuming that's what the builder had locally
 # ignores `The command '/bin/sh -c go get -d istio.io/istio' returned a non-zero code: 1`
-RUN go get -d istio.io/istio || true   
+RUN git clone https://github.com/istio/istio /go/src/istio.io/istio
 RUN cd /go/src/istio.io/istio && make istioctl
 
 ############################################
 # Main image
-FROM  kindest/node:v1.14.1
+FROM kindest/node:v1.15.0@sha256:b4d092fd2b507843dd096fe6c85d06a27a0cbd740a0b32a880fe61aba24bb478
```
Comment (costinm) [inline test/docker/Dockerfile]: They don't have a tag ?

Comment (howardjohn) [reply test/docker/Dockerfile]: `v1.15.0` is the tag. In the release notes the recommend adding the sha I guess its not needed though

File: test/docker/Dockerfile (hunk 1 above)
Comment (howardjohn) [reply test/docker/Dockerfile]: `build istio.io/test-infra/toolbox/ci2gubernator: cannot load github.com/google/go-github/github: unexpected status (https://proxy.golang.org/github.com/google/go-github/@v/v0.0.0-20171108000855-8c08f4fba5e0.zip): 400 Bad Request` if it is included. Not sure why

**Comment by howardjohn** [comment]
This will fix https://github.com/istio/installer/pull/146/files because the docker image has istioctl built into it so it was old. @qfel

File: test/docker/Dockerfile (hunk 1 above)
Comment (costinm) [reply test/docker/Dockerfile]: Did it fail once or consistently ? We should check what is the speed impact, and maybe look for a different proxy.

File: test/docker/Dockerfile (hunk 2 above)
Comment (costinm) [reply test/docker/Dockerfile]: I really don't like SHAs copied by hand. It's the job of tools.

File: test/docker/Dockerfile (hunk 1 above)
Comment (howardjohn) [reply test/docker/Dockerfile]: It is failing every time. I tried 8 days ago, 4 days ago and today. Not sure why though.. either way this should only effect the docker image being built, so speed is not a major concern right?

File: test/docker/Dockerfile (hunk 2 above)
Comment (howardjohn) [reply test/docker/Dockerfile]: I just removed it

**Comment by howardjohn** [comment]
blocked by https://github.com/istio/istio/issues/15432

**Comment by howardjohn** [comment]
Ah missed one.. Also blocked by https://github.com/istio/istio/pull/15463

File: test/docker/Dockerfile (hunk 1 above)
Comment (ericvn) [reply test/docker/Dockerfile]: I just tried pulling it locally this morning using the `GOPROXY=https://goproxy.io` and that worked fine. I have had issues even getting the _istio/istio_ modules to download with _https://proxy.golang.org_ (the later is still a Beta I believe).
